src/prognostic_equations/eddy_diffusion_closures.jl

```diff
@@ -612,6 +699,63 @@ function turbulent_prandtl_number(params, ᶜN²_eff, ᶜstrain_rate_norm)
     return min(max(prandtl_nvec, eps_FT), Pr_max)
 end
 
+
+function turbulent_prandtl_number(p)
+    (; params) = p
+    (; ᶜlinear_buoygrad, ᶜstrain_rate_norm) = p.precomputed
+    return @. lazy(
+        turbulent_prandtl_number(params, ᶜlinear_buoygrad, ᶜstrain_rate_norm),
+    )
+end
+
+
+"""
+    tke_exchange(Y, p)
+
+Calculates the turbulent kinetic energy (TKE) exchange tendency between the
+environment and updrafts due to detrainment.
+
+Arguments:
+- `Y`: The prognostic state vector.
+- `p`: Cache
+
+Returns:
+- The TKE exchange tendency term [m²/s³].
+"""
+function tke_exchange(Y, p)
+    (; turbconv_model) = p.atmos
+    n = n_mass_flux_subdomains(turbconv_model)
+    ᶜρa⁰ =
+        p.atmos.turbconv_model isa PrognosticEDMFX ? p.precomputed.ᶜρa⁰ : Y.c.ρ
+
+    ᶜtke_exch = p.scratch.ᶜtemp_scalar_2
```

Comment:
This can also be done lazily. Here it's a bit trickier because of the loop over `1:n`, but you can implement this as an `unrolled_map`. For example,
```
ᶜtke_exch_summands =
    unrolled_map(UnrolledUtilities.StaticOneTo(n)) do j
            Y.c.sgsʲs.:($$j).ρa * ᶜdetrʲs.:($$j) / ᶜρa⁰ * (
                1 / 2 * norm_sqr(ᶜinterp(ᶠu³⁰) - ᶜinterp(ᶠu³ʲs.:($$j))) -
                ᶜtke⁰
            )
    end
ᶜtke_exch = Base.Broadcast.broadcasted(+, ᶜtke_exch_summands...)
```
This is a little confusing to read though, so I can add a utility function that generalizes this for you.